homeassistant/components/ecobee/__init__.py

```diff
@@ -1,123 +1,162 @@
-"""Support for Ecobee devices."""
-import logging
-import os
+"""Support for ecobee."""
+import asyncio
 from datetime import timedelta
-
 import voluptuous as vol
 
-import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers import discovery
+from pyecobee import Ecobee, ECOBEE_API_KEY, ECOBEE_REFRESH_TOKEN, ExpiredTokenError
+
+from homeassistant.config_entries import SOURCE_IMPORT
 from homeassistant.const import CONF_API_KEY
+from homeassistant.helpers import config_validation as cv
 from homeassistant.util import Throttle
-from homeassistant.util.json import save_json
-
-_CONFIGURING = {}
-_LOGGER = logging.getLogger(__name__)
-
-CONF_HOLD_TEMP = "hold_temp"
 
-DOMAIN = "ecobee"
-
-ECOBEE_CONFIG_FILE = "ecobee.conf"
+from .const import (
+    CONF_HOLD_TEMP,
+    CONF_REFRESH_TOKEN,
+    DATA_ECOBEE_CONFIG,
+    DEFAULT_HOLD_TEMP,
+    DOMAIN,
+    ECOBEE_PLATFORMS,
+    _LOGGER,
+)
 
 MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=180)
 
-NETWORK = None
-
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
             {
                 vol.Optional(CONF_API_KEY): cv.string,
-                vol.Optional(CONF_HOLD_TEMP, default=False): cv.boolean,
+                vol.Optional(CONF_HOLD_TEMP, DEFAULT_HOLD_TEMP): cv.boolean,
             }
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
-def request_configuration(network, hass, config):
-    """Request configuration steps from the user."""
-    configurator = hass.components.configurator
-    if "ecobee" in _CONFIGURING:
-        configurator.notify_errors(
-            _CONFIGURING["ecobee"], "Failed to register, please try again."
+async def async_setup(hass, config):
+    """
+    Ecobee uses config flow for configuration.
+
+    But, an "ecobee:" entry in configuration.yaml will trigger an import flow
+    if a config entry doesn't already exist. If ecobee.conf exists, the import
+    flow will attempt to import it and create a config entry, to assist users
+    migrating from the old ecobee component. Otherwise, the user will have to
+    continue setting up the integration via the config flow.
+    """
+    hass.data[DATA_ECOBEE_CONFIG] = config[DOMAIN]
+
+    if not hass.config_entries.async_entries(DOMAIN):
```

Comment:
Add a check if the config isn't empty. If it's empty, we shouldn't import, since then we're triggered by the user in GUI.